runtime/ftplugin/query.lua

```diff
@@ -14,6 +14,9 @@ vim.treesitter.start()
 -- set omnifunc
 vim.bo.omnifunc = 'v:lua.vim.treesitter.query.omnifunc'
 
+-- recognize @foo.bar as a single word
+vim.opt.iskeyword:append('.')
```

Comment:
wouldn't that change user settiongs after they once opened the query editor?
and I think especially `.` is problematic because `word.another` is one of the biggest usecases of `w` imo

perhaps make this buffer local?

Reply:
This needs to be buffer-local, of course. I would recommend using vim.bo with an explicit list.

Reply:
In that case should I move this code to `dev.lua`? Else wouldn't it continue affecting all query buffers?

Reply:
I think the iskeyword should be consistent, unless that's not possible.

Reply:
Sorry if I'm being slow but how do I set it explicitly? Doesn't that require a nontrivial check like [this](https://github.com/neovim/neovim/blob/cbadb39d16c15b2a3d4f9a122644b3d61ba2d494/runtime/syntax/lisp.vim#L17C1-L23C6)? Why not just append the `.` to the existing `iskeyword` setting?

Reply:
`vim.bo` doesn't have the methods like `append`

Reply:
@max397574 I was thinking of just replacing line 18 by `vim.cmd([[ set iskeyword+=. ]])`

Reply:
Either look at the current iskeyword in a query buffer and hard code that list (with . added), or use vim.opt_local:append.

(The former is more future proof, but the latter is acceptable in a pinch.)

Reply:
would also work
not sure @clason do you have a convention of how `vim.cmd` is used? (indexing vs not indexing and [[]] vs "") 
Because I'd have used `vim.cmd.setlocal("iskeyword+=.")`

Reply:
@clason I would have thought that the former was more prone to breaking because you know, hardcoded stuff (not sure how often `iskeyword` changes though).

Reply:
Pretty much never, because that would bring out pitchforks....